src/internal/EnhancedButton.js

```diff
@@ -296,7 +296,7 @@ class EnhancedButton extends Component {
       display: 'inline-block',
       fontFamily: this.context.muiTheme.baseTheme.fontFamily,
       WebkitTapHighlightColor: enhancedButton.tapHighlightColor, // Remove mobile color flashing (deprecated)
-      cursor: disabled ? 'default' : 'pointer',
+      cursor: disabled ? 'not-allowed' : 'pointer',
```

Comment:
The why is *separation of concerns*.
I believe that the `cursor: not-allowed` is a visual helper that only make sense with a specific style.
The `<EnhancedButton />` component isn't responsible for changing his style based on the `disable` property.
The parent component his. Hence, I would rather set the `cursor` in the parent component.